app/user_views.py

```diff
@@ -94,6 +116,8 @@ def edit_rehearsal(rehearsal_id):
 def warm_up(rehearsal_id):
     form = RehearsalForm()
     requested_rehearsal = Rehearsal.query.get(rehearsal_id)
+    if requested_rehearsal.user_id != current_user.id:
```

Comment:
I wonder if there exists (or you could create) a function decorator which checks this instead of doing it inside the function.

This may be more of a "bonus" exercise (i.e. potentially a deep dive that you may or may not want to do now, or ever). The advantage is that a decorator is a little easier:
* To add to a function
* To see at a glance (without looking inside the function) that a function has such-and-such authentication requirements.

Reply:
Done and pushed!